### PR TITLE
TokenAllocation: allocation card

### PR DESCRIPTION
**Pull request by SzymonGos**

### CODE REVIEW

**Comment by SzymonGos** [comment]
As we discussed: I left the delete option unable for the first allocationCard. In my opinion, this will lead to a good UX.

File: pages/tokenAllocation.tsx [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
 import Container from '@/components/Container'
+import AllocationCard from '@/components/AllocationCard'
+import SecondaryButton from '@/components/buttons/SecondaryButton'
 
 export default function tokenAllocation() {
+  const [allocations, setAllocations] = useState([0])
+  const [counter, setCounter] = useState(0)
+
+  function addAllocation() {
+    setCounter((prev) => (prev === 14 ? 0 : prev + 1))
```
Comment (korzewski) [inline pages/tokenAllocation.tsx:11]: Import this array `colourPallete` and write it like this

`prev === colourPallete.length`? 0 : prev + 1

File: pages/tokenAllocation.tsx [hunk 2]
```diff
@@ -1,8 +1,25 @@
+import { useState } from 'react'
 import Container from '@/components/Container'
+import AllocationCard from '@/components/AllocationCard'
+import SecondaryButton from '@/components/buttons/SecondaryButton'
 
 export default function tokenAllocation() {
+  const [allocations, setAllocations] = useState([0])
+  const [counter, setCounter] = useState(0)
```
Comment (korzewski) [inline pages/tokenAllocation.tsx:8]: Not sure why you need this counter for?

File: pages/tokenAllocation.tsx [hunk 3]
```diff
@@ -1,8 +1,25 @@
...
+
+  function addAllocation() {
+    setCounter((prev) => (prev === 14 ? 0 : prev + 1))
+    let lastItem = allocations.slice(-1).toString()
+    setAllocations([...allocations, parseInt(lastItem) + 1])
+  }
+
+  function deleteAllocation(index: number) {
```
Comment (korzewski) [inline pages/tokenAllocation.tsx:16]: Please create token context and move this logic to separate file. Then you could get all the logic via 
```const {allocations, addAllocation, removeAllocation} = useToken()```

Same as useCart logic from the other project we've been doing

File: styles/index.scss [hunk 4]
```diff
@@ -5,6 +5,9 @@
 @import "./utilities.scss";
 @import "./fonts.scss";
 
+html {
+  background: #071741;
+}
 .body-gradient {
```
Comment (korzewski) [inline styles/index.scss:11]: Can we have empty line between new css classes?

File: components/Delete.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import React from 'react'
+
+export default function Delete(props: React.SVGProps<SVGSVGElement>) {
```
Comment (korzewski) [inline components/Delete.tsx:3]: Hover state doesn't look like the one on the designs

**Review by korzewski — CHANGES_REQUESTED** [review]
I assume the logic would come later, so not checking it now

File: components/AllocationCard.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        </div>
+      </div>
+      <Input label='Address' placeholder='hv423jl5v3h52v123klbn41klb14' />
+      {colour && (
+        <div
+          className='absolute top-0 bottom-0 m-auto left-[-8px] w-[16px] h-[16px] rounded-full'
+          style={{ backgroundColor: colour }}
+        ></div>
+      )}
+      {myKey === 0 ? null : (
```
[the remark this answers is not in the document]
Comment (SzymonGos) [reply components/AllocationCard.tsx]: I don't get why I should swap to boolean. In the end, I must use some kind of statement to verify which card is index === 0.

**Comment by SzymonGos** [comment]
FYI: I added useLocalStorage, however, during implementation I was receiving an error: 'Hydration failed because the initial UI does not match what was rendered on the server'. Right now is disabled, as this can be resolved later and I don't want to be stuck here.

**Comment by tk-o** [comment]
> FYI: I added useLocalStorage, however, during implementation I was receiving an error: 'Hydration failed because the initial UI does not match what was rendered on the server'. Right now is disabled, as this can be resolved later and I don't want to be stuck here.

This issue happens as localStorage is client-side only API. If you want to avoid hydration issues, you can use cookies (as they can be accessed on both, server-side, and client-side).

File: components/AllocationCard.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+      <Input label='Address' placeholder='hv423jl5v3h52v123klbn41klb14' />
+
+      {colour && (
+        <div
+          className='absolute top-0 bottom-0 m-auto left-[-8px] w-[16px] h-[16px] rounded-full'
+          style={{ backgroundColor: colour }}
+        ></div>
+      )}
+
+      {index && (
```
Comment (korzewski) [inline components/AllocationCard.tsx:42]: This line would print `0` on the first item. Please replace it with `!!index`

**Review by korzewski — APPROVED** [review]
Just minor comment, please merge after that

**Pull request merged**